rclcpp/src/rclcpp/executors/timers_manager.cpp

```
@@ -191,16 +244,29 @@ void TimersManager::clear()
   timers_cv_.notify_one();
 }
 
-void TimersManager::remove_timer(rclcpp::TimerBase::SharedPtr timer)
-{
-  this->remove_timer_raw(timer.get());
-}
-
-void TimersManager::remove_timer_raw(rclcpp::TimerBase * timer)
+void TimersManager::remove_timer(TimerPtr timer)
 {
   {
     std::unique_lock<std::mutex> timers_lock(timers_mutex_);
-    heap_.remove_timer(timer);
+
+    bool matched = false;
+    auto it = weak_timers_.begin();
+
+    while (it != weak_timers_.end()) {
+      matched = (it->lock() == timer);
+
+      if (matched) {
+        weak_timers_.erase(it);
+        break;
+      }
+
+      it++;
+    }
+
+    if ((it == weak_timers_.end()) && !matched ) {
+      throw std::runtime_error("Tried to remove timer not stored in the timers manager.");
```

I don't know about this. Do we want to throw an error in this case?
Noticed this because the timers manager test fails, it doesn't expect an exeption when removing a `nullptr`
https://github.com/irobot-ros/rclcpp/blob/irobot/add-events-executor/rclcpp/test/rclcpp/executors/test_timers_manager.cpp#L129